File: src/main/java/com/sequenceiq/cloudbreak/repository/HostGroupRepository.java
package com.sequenceiq.cloudbreak.repository;

import java.util.Set;

import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;

import com.sequenceiq.cloudbreak.domain.HostGroup;

public interface HostGroupRepository extends CrudRepository<HostGroup, Long> {

    Set<HostGroup> findHostGroupsInCluster(@Param("clusterId") Long clusterId);

    HostGroup findHostGroupInClusterByName(@Param("clusterId") Long clusterId, @Param("hostGroupName") String hostGroupName);

    Set<HostGroup> findAllHostGroupsByRecipe(@Param("recipeId") Long recipeId);

}

File: core/src/main/java/com/sequenceiq/cloudbreak/repository/HostGroupRepository.kt
package com.sequenceiq.cloudbreak.repository

import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param

import com.sequenceiq.cloudbreak.domain.HostGroup

@EntityType(entityClass = HostGroup::class)
interface HostGroupRepository : CrudRepository<HostGroup, Long> {

    fun findHostGroupsInCluster(@Param("clusterId") clusterId: Long?): Set<HostGroup>

    fun findHostGroupInClusterByName(@Param("clusterId") clusterId: Long?, @Param("hostGroupName") hostGroupName: String): HostGroup

    fun findAllHostGroupsByRecipe(@Param("recipeId") recipeId: Long?): Set<HostGroup>

    fun findHostGroupsByInstanceGroupName(@Param("clusterId") clusterId: Long?, @Param("instanceGroupName") instanceGroupName: String): HostGroup

}
